apps/dojo/src/agents.ts

```diff
@@ -25,451 +26,288 @@ import { A2AClient } from "@a2a-js/sdk/client";
-        }),
-        backend_tool_rendering: new ServerStarterAllFeaturesAgent({
-          url: `${envVars.serverStarterAllFeaturesUrl}/backend_tool_rendering`,
-        }),
-        human_in_the_loop: new ServerStarterAllFeaturesAgent({
-          url: `${envVars.serverStarterAllFeaturesUrl}/human_in_the_loop`,
-        }),
-        agentic_generative_ui: new ServerStarterAllFeaturesAgent({
-          url: `${envVars.serverStarterAllFeaturesUrl}/agentic_generative_ui`,
-        }),
-        tool_based_generative_ui: new ServerStarterAllFeaturesAgent({
-          url: `${envVars.serverStarterAllFeaturesUrl}/tool_based_generative_ui`,
-        }),
-        shared_state: new ServerStarterAllFeaturesAgent({
-          url: `${envVars.serverStarterAllFeaturesUrl}/shared_state`,
-        }),
-        predictive_state_updates: new ServerStarterAllFeaturesAgent({
-          url: `${envVars.serverStarterAllFeaturesUrl}/predictive_state_updates`,
-        }),
-      };
-    },
-  },
-  {
-    id: "mastra",
-    agents: async () => {
-      const mastraClient = new MastraClient({
-        baseUrl: envVars.mastraUrl,
-      });
+        // predictive_state_updates: "predictive_state_updates",
+        shared_state: "shared_state",
+        tool_based_generative_ui: "tool_based_generative_ui",
+        backend_tool_rendering: "backend_tool_rendering",
+      }
+    ),
 
-      return MastraAgent.getRemoteAgents({
-        mastraClient,
-      });
-    },
-  },
-  {
-    id: "mastra-agent-local",
-    agents: async () => {
-      return MastraAgent.getLocalAgents({ mastra });
-    },
+  "server-starter": async () => ({
+    agentic_chat: new ServerStarterAgent({ url: envVars.serverStarterUrl }),
+  }),
+
+  "adk-middleware": async () =>
+    mapAgents(
+      (path) => new ADKAgent({ url: `${envVars.adkMiddlewareUrl}/${path}` }),
+      {
+        agentic_chat: "chat",
+        agentic_generative_ui: "adk-agentic-generative-ui",
+        tool_based_generative_ui: "adk-tool-based-generative-ui",
+        human_in_the_loop: "adk-human-in-loop-agent",
+        backend_tool_rendering: "backend_tool_rendering",
+        shared_state: "adk-shared-state-agent",
+        // predictive_state_updates: "adk-predictive-state-agent",
+      }
+    ),
+
+  "server-starter-all-features": async () =>
+    mapAgents(
+      (path) => new ServerStarterAllFeaturesAgent({ url: `${envVars.serverStarterAllFeaturesUrl}/${path}` }),
+      {
+        agentic_chat: "agentic_chat",
+        // TODO: Add agent for agentic_chat_reasoning
+        // agentic_chat_reasoning: "agentic_chat_reasoning",
+        backend_tool_rendering: "backend_tool_rendering",
+        human_in_the_loop: "human_in_the_loop",
+        agentic_generative_ui: "agentic_generative_ui",
+        tool_based_generative_ui: "tool_based_generative_ui",
+        shared_state: "shared_state",
+        predictive_state_updates: "predictive_state_updates",
+      }
+    ),
```

Comment:
`"agentic_chat_reasoning"` was missing here but is expected in the list of agents for `"server-starter-all-features"` in `apps/dojo/src/menu.ts`.
https://github.com/ag-ui-protocol/ag-ui/blob/69320c98219eba641a0127287be520c080c3adb6/apps/dojo/src/menu.ts#L194-L207